src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -318,6 +319,14 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 		}
 	}
 
+	private getCurrentLinkAddress(): string {
+		if (document.getSelection().anchorNode.parentNode['protocol'] === 'file:') {
+			return document.getSelection().anchorNode.parentNode['pathname'] || '';
```

Comment:
Does getSelection also handle just having the cursor in the middle of the link (not selecting any text)? I would expect that even if I don't have anything selected it would still edit that link. 

The comment for the function indicates it does but I just want to make sure that's how it behaves.

And then if it does do that did you test what happens when that happens? Will it update the link as expected or does it try to insert a new link at that location?

Reply:
I believe that the intended scenario is that the user would select the entire link, as the link does get the link address wherever selected on the link. However, the display text does not include the title, which would require additional work.

Reply:
Not sure I understand - could you get a gif of the behavior for this? I don't think many people will know necessarily to select the entire text (and what happens if they select more/less) so we should probably spend some time now making sure we at least don't break stuff when that happens.

Reply:
Well, I went based on the issue that Julie shows in the [gif](https://github.com/microsoft/azuredatastudio/issues/14724). I think that is the clear scenario that we should address. Addressing the selection of the link regardless of where the cursor is and replacing the entire link based on any location of the link seems to be another piece of work entirely.

Reply:
I agree, I'm mostly saying make sure you test these other scenarios to verify that they aren't broken and then make sure you communicate expectations to the Livesite team so they know what exactly is supported.

Reply:
Julie also mentioned this in the issue:
"2. Users can place a cursor in the middle of the_Display Text_, and click on the Link button, then show the current values for both Text to display and Address (currently only shows Text to Display value). Changing the Text to display and Address values, will update the Display Text and the URL."
So it would probably be good to check with her or create a new issue to track the second part.

Reply:
"2. Users can place a cursor in the middle of the_Display Text_, and click on the Link button, then show the current values for both Text to display and Address (currently only shows Text to Display value). Changing the Text to display and Address values, will update the Display Text and the URL."

I would consider this as the next step (incremental). It is possible to ship without it for now.